pkg/webhook/virtualmachine/virtualmachine.go

```diff
@@ -98,19 +100,26 @@ func (a *virtualMachineAnnotator) Handle(ctx context.Context, req admission.Requ
 func patchVMChanges(originalVirtualMachine, currentVirtualMachine *kubevirt.VirtualMachine) admission.Response {
 	var kubemapcoolJsonPatches []jsonpatch.Operation
 
-	for ifaceIdx, _ := range currentVirtualMachine.Spec.Template.Spec.Domain.Devices.Interfaces {
-		interfacePatches, err := patchChange(fmt.Sprintf("/spec/template/spec/domain/devices/interfaces/%d/macAddress", ifaceIdx), originalVirtualMachine.Spec.Template.Spec.Domain.Devices.Interfaces[ifaceIdx].MacAddress, currentVirtualMachine.Spec.Template.Spec.Domain.Devices.Interfaces[ifaceIdx].MacAddress)
+	if pool_manager.IsNotVirtualMachineMarkedForDeletion(currentVirtualMachine) {
+		if pool_manager.GetTransactionTimestampAnnotationFromVm(originalVirtualMachine) != pool_manager.GetTransactionTimestampAnnotationFromVm(currentVirtualMachine) {
+			transactionTimestampAnnotationPatch := jsonpatch.NewPatch("add", "/metadata/annotations", pool_manager.GetTransactionTimestampAnnotation(currentVirtualMachine))
```

Comment:
I wanted to be specific, but you're right there's no harm in using `patchChange`

Reply:
mmm.. the problem is that the patch generated is not a map[string]string, but only the value:
```
{"op":"add","path":"/metadata/annotations","value":"2021-02-15T10:13:11.760428025Z"}
```
instead of 
```
{"op":"add","path":"/metadata/annotations","value":"kubemacpool.io/transaction-timestamp:2021-02-15T10:13:11.760428025Z"}
```